lib/ansible/modules/system/mount.py

```diff
@@ -725,6 +726,14 @@ def main():
             module.fail_json(msg="Error mounting %s: %s" % (name, msg))
     elif state == 'present':
         name, changed = set_mount(module, args)
+    elif state == 'remount':
+        res = 0
+
+        res, msg = remount(module, args)
+        changed = True
+
+        if res:
+            module.fail_json(msg="Error remounting %s: %s" % (name, msg))
```

Comment:
Move this condition above the `changed = True` and inside the `if module.check_mode:`.